geode-core/src/main/java/org/apache/geode/pdx/internal/TypeRegistrationStatistics.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.pdx.internal;
+
+import org.apache.geode.StatisticDescriptor;
+import org.apache.geode.Statistics;
+import org.apache.geode.StatisticsFactory;
+import org.apache.geode.StatisticsType;
+
```

Comment:
With a new class, we have an opportunity to name the instrumentation methods to focus on reporting stuff, rather than on manipulating statistics. For example: Something like `pdxTypeDefined()` and `pdxTypeCreated()`.

Similar for the class name… Is there a good name that focuses on reporting occurrences of interesting events, rather than on statistics? I don't have a great suggestion for what to name the class. My not-necessarily-great attempt is `TypeRegistrationReporter`.

Reply:
I kind of like this idea but what other sort of stuff would we put in these methods? It wouldn't make sense for them to write to the log. So outside of stats, what are you thinking?

If we limit it to stats I still think changing the method names is a reasonable ask. It certainly makes it more readable.

Reply:
I left the class name the same but did change the methods to be more readable. Thoughts?

Reply:
My hope for the future is this: We can hide a certain amount of logging and measuring in the same instrumentation class. In my future fantasy world, the code reports an event to the instrumentation class, and the instrumentation class logs, measures, or both.

A suffix more general than 'stats' might invite us to put some logging in there. The best I can think of is 'Instrumentation' (as in some of the examples from [Pete Hodgson's instrumentation article](https://martinfowler.com/articles/domain-oriented-observability.html)).

I'd like to move in that direction at some point. I recognize that shifting the style will put an extra burden on whoever writes the first few new instrumentation classes.

So given all of that (non-blocking) input, do what you think best.

Reply:
I think out of scope for this because we really need a proposal and examples to drive this. I don't want to trailblaze this.